### ISSUE TITLE
Homebridge running but “no response” in HomeKit 

### ISSUE THREAD

**Issue by jeffl78** [homebridge/homebridge#1801]
The title pretty much says it all but this seems to be happening more often since I updated to iOS 11.3 on the iPhone X and 11.3.0 on the Apple TV 4th generation.  I have tried deleting persist and accessories, resetting Apple TV, upgrading homebridge, adding and removing accessories in HomeKit.  After repeating this a few times I started removing homebridge completely and reinstalling, which I tried a few times.  Homebridge will run just fine but after a while, say an hour or so, homebridge and accessories stop responding in HomeKit.  After a reboot homebridge will once again, work for about an hour.

Any help would be appreciated.

- **Homebridge**: `0.4.38`
- **Node**:      `9.10.1`
- **NPM**:    `5.6.8`
- **OS**:         Raspberry Pi 3b with Rasbian and Noobs

**Comment by paki81** [comment]
Same issue for me after update my device to 11.3

**Comment by cmcken** [comment]
Also same issue for me after update to 11.3

**Comment by davidvanvo** [comment]
has there been any changes to the network or to the raspberry pi? i’m on a mac installation with the latest tvos and ios and all of our phones have been working fine.

**Comment by cmcken** [comment]
No network changes - only change I am aware of is update to ios 11.3. I am also running on a raspberry pi.

**Comment by Veldkornet** [comment]
I have no problems either...

Node: 8.9.4
NPM: 5.6.0
homebridge: 0.4.38
OS: Raspbian (RaspberryPi 2)

**Comment by paki81** [comment]
No network change, only devices updated to last ios and tv os

**Comment by paki81** [comment]
Can you tell us how you solve the problem?

**Comment by mbdmbd** [comment]
FWIW, I was following this as I was having similar issues over the past couple of days (running on a mac). I noticed that Little Snitch was blocking connections to node due to an 'invalid code signature'. I was able to confirm this by testing the configured port (as defined in the config.json). After forcing the rule in Little Snitch to allow connections, and a successful tcp connection was possible from devices on my network, I was good to go again

`➜  nc -z 192.168.1.88 51826`
`Connection to 192.168.1.88 port 51826 [tcp/*] succeeded!`

**Comment by jeffl78** [comment]
I reformatted my pi, reset my apple TV, and re-added everything and my HomeKit accessories work for a about 10 hours or so before I get "no responce" from them after which a reboot seems to reproduce this.  I am trying to figure out what mbdmbd did and if it applies to my situation but otherwise I am running out of things to try.  I have a feeling on of my plugins may be causing the issue.  Any input is certainly welcome.

**Comment by mbdmbd** [comment]
In my case it was effectively a network issue where Little Snitch (a Mac App that acts as a kind of firewall to stop your machine from sending/receiving unexpected data) had some issues with node and was blocking incoming and outgoing tcp connections. So, it may not be the exact same issue, but it may be worth checking whether or not you're able to access homebridge on the TCP port of the RPI from other network hosts. (and vice versa).

**Comment by andyoneal** [comment]
same issue here. not running little snitch. also running off of a raspberry pi 3.

**Comment by we5** [comment]
In my experience there is usually some problems with the platforms or accessories which messes up HomeKit. Unfortunately sometimes even the 3rd party apps like Home or Eve can not unblock it.

I am running FHEM with homebridge-fhem and I keep getting problems or unreliable behavior if the homebridge-fhem platform plugin gets more than 50 devices. I now try to split this up to multiple platforms or multiple homebridge instances. I already have one separate instance for a guest Home and this is working just fine, even if there are the same "problematic" devices (but a lot less than 50 of them).

**Comment by jungopro** [comment]
Hi All
I'm having the exact same issue. Mine Homebridge is running on:

- Levnovo Yoga with Windows 10
- iphone 8 with latest iOS

In my case, restarting homebridge fixes the issue for a few minutes and then it returns and status is no response

**Comment by jungopro** [comment]
FYI, I installed homebridge inside ask on Windows and getting the same issue...

Anyone? Anything???

**Comment by tosate** [comment]
I am having the same issue on  an iPad and an iPhone running iOS 11.3. These devices state that there is "no response" in Homekit. I have another iPhone that is still on iOS 11.2.6 and my homebridge instance is working fine on this device.

**Comment by jungopro** [comment]
So it's an iOS issue I guess.. funny, I'm testing [HA](https://www.home-assistant.io) with the HomeKit plugin in parallel and I don't experience the same issues there...

**Comment by Squuiid** [comment]
"Homebridge will run just fine but after a while, say an hour or so, homebridge and accessories stop responding in HomeKit. After a reboot homebridge will once again, work for about an hour."

Same issue here. Running on Ubuntu Server 16.04.4 LTS VM.

**Comment by jeffl78** [comment]
I got mine running fairly well now after resetting everything I listed above.  Everything works fine for days at a time, the same “no response” in HomeKit issue only happens and requires a reboot to correct every few days.  I’ll try to get more details about the logs and post them when I can.

**Comment by Squuiid** [comment]
@jeffl78 Can you post what versions of everything you are now running with it working? This would be very helpful. Also, are you on iOS 11.3 on all of your devices?
Thanks.

**Comment by Squuiid** [comment]
UPDATE 2; Nope. Issue persists when using Node.js 8.x. It just took a little longer to become unresponsive.
This is very frustrating.

**Comment by cubilon** [comment]
let’s try node.js 1.0
just kidding 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Erokani** [comment]
Hi,
Same thing, hombridge running on debian jessie, devices with iOS 13.
For me, when it happens, I need to change the rooms in the homekit app until it finally ends recovering the connection. I don't need a reset. However this is very frustrating to wait until it succeeds...

**Comment by Squuiid** [comment]
It turns out that my problem was that the plugin I was running crashes Homebridge.
The only plugin I run is HarmonyHub and it is incredible unstable. Dev seems to have abandoned the project so there’s little hope of things improving unfortunately.

**Comment by HelloNewW** [comment]
Thanks for your guidance... 
It`s looks like that I need to load hombridge in debug mode and make some research.
Possibly one or more of my plugins unstable with the new IOS11.3 
Hope

**Comment by tosate** [comment]
As I have already mentioned, I have an iPhone which is still running on iOS 11.2.6 and another iPhone which runs iOS 11.3. Homebridge works perfectly with the first device. On the iOS 11.3 device I get the 'no response' error.
I have used tcpdump and wireshark to find out what the two devices do when I start the home app. In both cases I have killed the home app on the device and restarted the homebridge daemon. After that I have recorded the network traffic on the homebridge port and I have started the home app.

Disclaimer: I have no knowledge about HomeKit or homebridge, therefore I can just describe what I see in the network traffic.

[iOS 11.2.6]
The iOS device sends a request to the homebridge server. It's a raw TCP packet with probably encrypted data. I cannot recognize any familiar strings in the data. Homebridge replies with three acknowledge [ACK] packets, followed by a [PSH, ACK] packet. The iPhone responds with two [ACK] pakets. Everything happens in 7 TCP packets going back and forth.

[iOS 11.3]
The iOS device sends a synchronization request [SYN] which is acknowledged by the homebridge server with a [SYN, ACK] request. The iPhone also acknowledges the synchronization with an [ACK] packet. After that the iPhone performs a pairing over HTTP (media type: application/pairing+tlv8). The homebridge server replies with HTTP 200 OK and the iPhone acknowledges this packet. Then the iPhone sends another pairing request over HTTP and the homebridge server replies with HTTP 200 OK.
After those two pairing request (which have failed probably), the iPhone stops the connection with a FIN packet.
Then it all starts from the beginning and this sequence is repeated 5 times.

So what is the problem? Unfortunately, I don't know. Maybe the pairing mechanism was not enforced in previous versions of iOS and is mandatory now. I could not find the string 'tlv8' in the homebridge repository, so my guess is that this protocol is not implemented yet by homebridge. Maybe I am completely wrong.

A copy of the "HomeKit Accessory Protocol Specification" describing the pairing over IP can be found at https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=4&ved=0ahUKEwjpiJmtxt_aAhVGFywKHQ6xCgYQFghGMAM&url=https%3A%2F%2Fcommunity.ubnt.com%2Fubnt%2Fattachments%2Fubnt%2FairVision_Ideas%2F2505%2F1%2FHAP-Specification-Non-Commercial-Version.pdf&usg=AOvVaw07maQN8gvW6VeZbamuoJux

Another HomeKit HAP implementation with implemented tlv8 is https://github.com/etwmc/Personal-HomeKit-HAP

Hopefully this will help to fix this issue.

Cheers,
tosate

**Comment by HelloNewW** [comment]
Hi tosate ,
thank you for sharing your research.
I notice that my homekit work perfectly.
10ms until homekit recognize all my device with command delay.
The problem is not systematically .
Thousand of users over the world use homebridge plugin on IOS 11.3. 
It’s really strange that we don’t see thousand of comment about that issue.
It looks like it depends on fault(unsupported) plugin combination.

**Comment by HelloNewW** [comment]
Sorry
*10ms until homekit recognize all my device without  command delay issues .

**Comment by Erokani** [comment]
Yep. Still... :(

**Comment by djrobx** [comment]
I had this problem too but I got it to work again.   I previously had a working system until I updated Homebridge and Node because the nest plugin was complaining about some responses.    Unfortunately I don't remember what the old version, just that it was before the homebridge output had a QR code (only the numeric pairing code).   I tried a few different versions of Node.   The latest Node had a strange problem of saying the port was already in use.   But no errors with Node8. 

After the updates, my devices could not find homebridge ("No response)".   I loaded Bonjour Browser on my mac, and saw that _hap._tcp was not showing up.    Other avahi-daemon services on the same box were showing, however, so avahi itself was running (unsurprising, I did not update that).

In trying to learn how to troubleshoot avahi, I found its configuration files have the ability to do hardcoded services (which is where the *working* entries were coming from).  So I tried to force a _hap._tcp.    This did show up in Bonjour Browser but did not work to pair Homekit.      I took a look again at the homebridge router side by running

avahi-browse -avr

Now that I added the "fake" entry, I can ALSO see Homebridge's entry (on a different interface IP address for some reason?!).  Before this did not showe up.

=    em1 IPv4 router                                        _hap._tcp            local
   hostname = [router.local]
   address = [192.168.0.1]
   port = [28000]
 
=    em0 IPv4 Homebridge-65E3                               _hap._tcp            local
   hostname = [C9_32_3D_E3_CE_42.local]
   address = [10.122.64.163]
   port = [28000]
   **txt = ["sh=eIiy3x==" "sf=1" "ci=2" "ff=0" "s#=1" "c#=2" "id=xx:xx:3D:E3:CE:42" "pv=1.0" "md=Homebridge"]**

 So armed with this, I went back into my fake entry and added all the txt entries:

**vi /usr/local/etc/avahi/services/hap-tcp.service**

<service-group>

  <name replace-wildcards="yes">%h</name>

  <service>
    <type>_hap._tcp</type>
    <port>28000</port>
    <txt-record>sh=xY+kcg==</txt-record>
    <txt-record>sf=1</txt-record>
<txt-record>ci=2</txt-record>
<txt-record>ff=0</txt-record>               
<txt-record>s#=1</txt-record>
<txt-record>c#=3</txt-record>
<txt-record>id=xx:xx:3D:E3:CE:42</txt-record>
<txt-record>pv=1.0</txt-record>
<txt-record>md=Homebridge</txt-record>
  </service>

</service-group>

Restarted avahi and now everything works.   

So in my case there is clearly some kind of breakdown in how Homebridge activating the service in Avahi that manually publishing it seems to work around.    The homebridge log shows no errors other than that typical Avahi deprecation warning.

**Comment by Nicuz** [comment]
Same issue here. After a node, npm and homebridge update I have no response by every accessories even if homebridge and all the plugins are running.

* Homebridge 0.4.42
* Node 8.9.0
* NPM 6.1.0
* Raspbian Jessie on Raspberry Pi 1 Model B

**Comment by Hucksleyy91** [comment]
I just installed a accessory and this started happening !
This has happened before !
To fix it last time and this time also was simple 

Don’t input ip address as “192.168.1.68”
I use “http://192.168.1.68”

Quick reboot of homebridge and problem solved :) hope it helps

**Comment by Erokani** [comment]
Wow. If this is as simple as that, I just die 😶😓😅
I’ll try asap.
I don’t have anymore homebridge conf file on top of my minds: which IP setting does it correspond to?
Thanks

**Comment by HelloNewW** [comment]
Hucksleyy91 
Where exactly to input that ip “http://xxx.xxx.x.xxx” ? 
in my config.json ?

**Comment by gizmo8500** [comment]
I have this same issue. I noticed on the main home bridge readme it mentions something about users getting "stuck" as inactive in the database, and to simply change the username and restart home bridge. This sounds to me like it could be the culprit of the issue we are all describing. Unfortunately changing the user's number is not very practical since then you have to add home bridge to HomeKit again, and then reconfigure all devices from scratch (this is quite time consuming for the 80+ devices I have). Is it possible for us to get a better solution?

**Comment by tdoubleyoo** [comment]
Same problem here - I noticed that disconnecting/reconnecting wifi on my iPhone solves the problem temporarily - same for you? I'm analyzing mDNS service publication/vanishing right now - same observations as djrobx.

Unfortunately djrobx' workaround didn't work for me. Both services show up, later both services vanish again.

**Comment by ebaauw** [comment]
I’ve noticed that too: switching WiFi off and on often remedies the _No response_ situation.  Sometimes it requires a restart of the iOS device, sometimes it remedies spontaneously after a couple of minutes.

I also noticed that sometimes my iPad shows _No response_ while my iPhone happily interacts with homebridge.  I’ve even noticed occasionally that the Home app shows _No response_ while Eve, on the same device, is able to interact.

I run two homebridge instances at home, one on a Raspberry Pi, one on a mac mini server.  Sometimes the _No response_ is shown for both instances, sometimes only for one, mostly the one running on the Raspberry.

Finally, there’s a lot of _No response_ complaints on the Hue developers forum, where they use the Hue bridge’s native HomeKit functionality instead of homebridge.

My conclusion is that this a not so much an issue of homebridge or the server running homebridge, but more an issue of the (HomeKit runtime on the) iDevice and/or the connection between the iDevice and the HomeKit accessory.  Maybe specific to HomeKit bridges with many bridged accessories.

**Comment by tdoubleyoo** [comment]
OK, solved it for my setup - like in many other cases, it was a router issue. In my case (AVM Fritzbox 7490) a really annoying one, since it handles multicast almost correct (i.e. IPTV works well). But it somehow determines in a non-standard way which packets to forward to which interfaces/devices. And this seems to be buggy. When I connect my raspberry pi running homebridge to a lan interface, it never works. pi & iPhone on wifi break sporadically (like described above, "not responding" until wifi deactivate/reactivate). I put pi & iPhone on wifi of an old router, works like a charm. 

Unfortunately AVM does not offer the option of shell access to its devices anymore and hacks for my specific device seem to be at least a bit experimental. I'll probably simply switch routers and try to forget this annoying episode :-)

**Comment by Stevy92** [comment]
I experienced the same problem. I updated npm, node.js and homebridge, but with no results. 
Then I read this threat and tried another router. This solved the problem. So, I updated my FritzBox 7430 with the labor version. This seems to solve the problem. Now my FritzBox is running again and homebridge is for now available. Thanks for this hint!

**Comment by duculete** [comment]
I have this problem also. after few hours it stops responding....

Aug 23 07:37:54 homebridge homebridge[6573]: Thu, 23 Aug 2018 04:37:54 GMT EventedHTTPServer [::ffff:192.168.1.47] New connection from client
Aug 23 07:37:54 homebridge homebridge[6573]: Thu, 23 Aug 2018 04:37:54 GMT EventedHTTPServer [::ffff:192.168.1.47] HTTP server listening on port 57789
Aug 23 07:37:54 homebridge homebridge[6573]: Thu, 23 Aug 2018 04:37:54 GMT EventedHTTPServer [::ffff:192.168.1.47] HTTP request: /pair-verify
Aug 23 07:37:54 homebridge homebridge[6573]: Thu, 23 Aug 2018 04:37:54 GMT HAPServer [CC:22:3D:E3:CE:30] HAP Request: POST /pair-verify
Aug 23 07:37:54 homebridge homebridge[6573]: Thu, 23 Aug 2018 04:37:54 GMT HAPServer [CC:22:3D:E3:CE:30] Pair verify step 1/2
Aug 23 07:37:54 homebridge homebridge[6573]: Thu, 23 Aug 2018 04:37:54 GMT EventedHTTPServer [::ffff:192.168.1.47] HTTP Response is finished
Aug 23 07:37:54 homebridge homebridge[6573]: Thu, 23 Aug 2018 04:37:54 GMT EventedHTTPServer [::ffff:192.168.1.47] Client connection closed
Aug 23 07:37:54 homebridge homebridge[6573]: Thu, 23 Aug 2018 04:37:54 GMT EventedHTTPServer [::ffff:192.168.1.47] HTTP connection was closed
Aug 23 07:37:54 homebridge homebridge[6573]: Thu, 23 Aug 2018 04:37:54 GMT EventedHTTPServer [::ffff:192.168.1.47] HTTP server was closed

**Comment by Erokani** [comment]
The previous suspicions of network handshake are kind of confirmed for me.
I recently bridged my Netatmo welcome camera via the ffmepg plugin, and found both the camera screenshot immediately & systematically working, while the timeout issues are not present any more for the other homebridge devices ONLY IF I first look at the camera... They are obviously different network protocols involved, and the camera ones helping the others to initiate.

**Comment by lamjoey** [comment]
I was having the no response problem, the issue for me was that I have a device which I would sometimes turn it off from the wall plug, with that single device not responding all of the accessories also become no response.  My solution was put this device that I would turn off from wall plug into another instance of homebridge. So now I have two instances and my other accessories would not be affected when I turn the device off.

**Comment by stateless-01** [comment]
@Stevy92  I have been suspecting my Fritzbox to be one of the problems regarding these issues, too.
What do you mean with ‚labor version’ of the FritzOS?

I have also found out that switching off the IGMP Snooping on my Netgear switches contributes to a more reliable connection of the Homebridge-App to the raspberry as this has also driven me crazy before.
Still I have the common problems with unreachable items in HomeKit.

**Comment by ebaauw** [comment]
>  I have been suspecting my Fritzbox to be one of the problems regarding these issues.

Interesting theory.  There's no IGMP Snooping setting on my Fritz!Box (7369, Fritz!OS 06.32).  I have a fairly large home network, with the Fritz!Box as central switch.  It connects to three WiFi access points (I don't use the WiFi of the Fritz!Box) and to a switch for my wiring closet (connected to my Mac mini server and Raspberries running homebridge).  I suspect the _Not responding_ issues occur especially, when my iPad or iPhone connects to a different access point.

To test this theory, I re-wired my network, so the Fritz!Box is no longer the central switch.  In other words: traffic between WiFi-connected devices and homebridge no longer runs through the Fritz!Box.  HomeKit seems way more responsive, when opening the Home app the _Updating..._ takes considerably less time, and I haven't seen a _Not responding_ state yet.

**Comment by stateless-01** [comment]
@ebaauw Hello and thanks for your input. How is your setup performing so far?

I can’t cut the fritzbox completely in my setup at the moment as it is vital for the telephones in our building. I could of course create a separate environment for the Homebridge.

I am also not using the built in WiFi of the router,  I am using multiple AP which are connected via ethernet. As the raspberry (Homebridge) and AppleTv (HomeKit hub) are also connected to the same network via Ethernet I am wondering how the router or even switch might EXACTLY cause the availability errrors we’re discussing here. 

I’d be willing to get any other router for this but then I need to know what configurable features to look for.

Homebridge is such a great thing that I am happy to figure this network issue out.

**Comment by ebaauw** [comment]
It seems to have solved the issue.  Sometimes the Home app is still stuck on _Updating_, but simply stopping and restarting the app seems to solve that.  No need to turn off and back on the iPhone’s WiFi, nor to reboot the iPhone.

You don’t need a new router.  The trick is to make sure the homebridge server(s) and HomeKit clients (iDevices) are all on the same network port of the Fritz!Box.  I’m still using my Fritz!Box for Internet access, I just have a single Ethernet cable from it to my newly designated master switch.  This is just a simple, unmanaged, consumer-grade, 8-port gigabit switch.

The issue is (presumably) caused by the router/switch trying to be too smart.  It keeps track of the multicast subscriptions by connected devices (IGMP snooping), and consequently forwards multicast traffic only to those ports that have a subscribed device.  If the device changes ports (because it switches access points, moving around the house), the router/switch needs to see a new multicast subscription before the traffic is forwarded to the other port.  However, the device thinks it’s still subscribed, and won’t issue this subscription for some time.  Until then, effectively, it doesn’t see the HomeKit accessory on the network.  By disabling IGMP snooping, the router/switch forwards multicast traffic to all ports.

You might also want to check that your access points don’t do IGMP snooping, especially if these are consumer-grade routers in bridge mode.  My 802.11ac Airport Extreme has an option for this, my older 802.11n Time Capsule doesn’t, but presumably doesn’t support IGMP snooping at all.

**Comment by stateless-01** [comment]
The situation where the HomeKit app has to be restarted once to be online again is exactly where I have been from the beginning. And this is what I am trying to solve.

I am only using one eth port of the fritzbox. From there it goes into a 24 port managed switch from Netgear. I have switched off IGMP Snooping on the switch, even though it doesn’t make a lot of sense to me a for the rest of the ethernet traffic. Switching it off, it has become a lot easier to access the homebridge server through the homeBridge app. But the other thing is still not working correctly. I want to get the Home app working perfectly without having to restart it every once in a while.

I am not the only person living in the house... ;-)

**Comment by ebaauw** [comment]
>  I want to get the Home app working perfectly without having to restart it every once in a while.

Same here!

> The situation where the HomeKit app has to be restarted once to be online again is exactly where I have been from the beginning. And this is what I am trying to solve.

I'm afraid this might very well have a different cause: a bug in the Home app, or in iOS' HomeKit runtime.  Sometimes I find the Home app is unresponsive, while at the same time, on the same device, Eve does respond.  HomeKit is very complex under the hood, and it's very difficult to figure out what's actually happening.

**Comment by stateless-01** [comment]
If I simply use Siri to execute something related to HomeKit, most of the time it works right away, no matter if I am logged into my home network or from the outside via the iCloud/Apple TV connection.

**Comment by djrobx** [comment]
@ebaauw, @stateless-01 

One side effect of the "hack" I did to get mine working again by "hardcoding" the service into Avahi (details posted earlier) is that it has completely stopped the intermittent failures.   Since doing that, it has had no issues in 3 months and now always works right away after reboot.   I used to have to periodically restart the service as you describe and often had to restart the service once following a reboot.    You might give it a try.

**Comment by ebaauw** [comment]
@djrobx, thanks for the tip.  On my Raspberry Pi, the file would be in `/etc/avahi/services`.  I already created an `rfb.service` there, to have my Pi's VNC server show up in macOS screen sharing.

I do see a potential issue with your hack:  The `c#` text record is the configuration version, which increments every time the configuration changes, see https://github.com/KhaosT/HAP-NodeJS/blob/9eaea6df40811ccc71664a1ab0c13736e759dac7/lib/Advertiser.js#L40.  I don't know the details, but it seems this is needed to inform HomeKit clients of new, changed and/or deleted accessories, services, and/or characteristics.

**Comment by Giltechy** [comment]
Hi guys,

Need some help on homebridge plugin, I have raspberry pi 3b + running on jessie. i have installed homebridge with 2 plugins smarthings and yeelight. when ever power out apcupsd shutdown the raspberry pi and once power resume apcupsd start the Pi. Everything working fine till this,  but once the pi started , the homebridge plugin yeelight running , but the smarthings not running.

any reason for this, help needed.

**Comment by Stevy92** [comment]
Sorry for wrong information.
For me it looked like the labor version (its the beta version from avm) solved it, but it didn't. I read something about, that setting the speed of the wifi to 54mbit helps, and indeed on my zeroconf browser on Android it did. Together with the Avahi "hack" my homebridge is visible all the time. But for the iPhone and homekit it is still the same. 90% of the time no response. It is really a pitty, that avm blocked shell access, so there is really limited stuff we can do. I will order a simple wifi router and just connect it behind the fritz box, because I need the fritz box for phone and my ISP don't want to share the password.

**Comment by ebaauw** [comment]
In theory, Apple could have changed the way iOS 12 handles multicast subscriptions so that IGMP snooping no longer causes issues.  I’m a bit reluctant to re-wire my network again, now that it’s working.  I could try enabling IGMP snooping on my Airport Extreme, and see how that works out.

**Comment by skywalker1979** [comment]
anyone running on UBIQUITI unifi access points? i think that is the problem for some reason.

**Comment by gizmo8500** [comment]
I am, yes. Ubiquiti access points ad unifi gateway.

Although since upgrading to iOS 12, like tdoubleyoo, I haven’t had it fail yet...could just be wishful thinking.


Bryan

**Comment by skywalker1979** [comment]
well...if it start happening, think about trying to downgrade to previous controller

**Comment by iandronowicz** [comment]
Im getting a little frustrated here... can't get it to work, it stopped working when I added a switch to manage several devices on my network (raspberry pi still using wifi). The log from Homebridge shows everything running ok but on the home app I get no response from all accessories. The strange thing is that I have another instance of Homebridge running on my Mac mini and it works perfectly.

**Comment by skywalker1979** [comment]
1- put the raspberry not on wifi but on ethernet....
2- upgrade to ios12
3- check your wifi access point settings.

**Comment by iandronowicz** [comment]
1-used to work on wifi, change it to ethernet and stopped working, now I reverted it to wifi but still no luck
2-done that when changing from wifi to ethernet
3-already on ios12 since launch
4-I have a 2 node Linksys Velop (installed it last week) y checked every option available, even disabled all "firewall" settings but still no luck.

Thanks for sharing... Right now uninstalled everything: Homebridge, plugins, purged node modules. Don't know which version of node to use... on my Mac mini homebridge (working perfectly) I have node 10.5.0

**Comment by iandronowicz** [comment]
well... re installed everything, and still getting "no response" on all accesories. even though on the terminal the output of Homebridge shows everything working just fine

**Comment by aliceforware** [comment]
I am have the same issue on my iPad and iPhone running iOS 11.4.1, I used iPad as HomeKit center, I got “no response”  in my iPad sometimes even I am using HomeKit app at the same iPad and all functions are working fine,I have been thinking it might be an issue on home center.

> I am having the same issue on an iPad and an iPhone running iOS 11.3. These devices state that there is "no response" in Homekit. I have another iPhone that is still on iOS 11.2.6 and my homebridge instance is working fine on this device.

**Comment by stateless-01** [comment]
Hello all,
you can see my earlier posts, I also had issues with non (or too slow) responding items, in my case only Homematic items.

Since updating all devices to iOS 12 final version I had very few problems, it seems to have made a major improvement in availability.

I was also able to switch on the IGMP snooping on my managed switches again. The deactivation had previously improved the availability of my raspberry running Homebridge. For some reason this makes no difference anymore since ios12.
I still believe that a change of my router (Fritzbox) could make another difference. I do have a few routers here I could use for a test setup. I simply didn’t have the time so far.

What I can say is: As long as I am only using Siri commands to do something with HomeKit/Homebridge, it works pretty much all the time now! If I launch the HOME app on an iOS device, sometimes i get the unavailable message, and after opening it again right away, everything works until I leave the network.

**Comment by peer69** [comment]
I have the same "no response" error. I've tried quite some things, even changed the switch to some enterprise grade model to use advanced igmp features. Didn't do anything to get homebridge work reliable. 

Which platforms are you guys using? I'm sensing that the error on my side has something to do with plugins sending querys to servers outside my home network (in my case the netatmo weather station plugin). I should have done this before but for now after about an hour of testing, deactivating the netatmo plugin has done the trick. All devices are responding as expected. This may just be a coincidence but homekit was incredible unresponsive for most of the time today so at least I feel some impact right now. Let's see how this works out.

**Comment by aliceforware** [comment]
I had set static IP for my homekit control center( my IPAD ) three days ago, it works fine now, keep observing

**Comment by henkhb** [comment]
Also on Fritzbox and with similar problem. 
Home app is refreshing only when switching rooms. After opening it shows no response most time.

**Comment by Stevy92** [comment]
After my wife updates to new ios 12.1 even with extra TP Link WiFi Router it stopped working. Really annoying. What is Apple doing there? I'm going to change TP Link firmware to DD-WRT and play with the settings.

**Comment by chrismaurer** [comment]
Are folks here still having issues? I just went through the same problem following a power outage and it was stressing me out to no end having put a lot of effort into getting my setup just right.

Here’s what worked for me:

 - Delete HomeBridge from HomeKit
 - Re-add HomeBridge to HomeKit
 - Sign out of iCloud on my iPad then sign back in

What “fixed” the issue was really deleting Bridge and re-adding it, but HomeKit somehow doesn’t recognize that it’s a new bridge (as evidenced by the fact that trying to re-add it without deleting the ./.homebridge/persist folder causes an “already added” error). Once Bridge was re-added successfully, it worked on the iPhone that I used to set it up, but nowhere else. My iPad acts as one of the backup HomeKit hubs and couldn’t sync this properly so the iCloud reset fixed that. This somehow fixed HomePod and AppleTV as well.

Good Luck!

**Comment by henkhb** [comment]
For me it was an old and not updated plugin. After removing that, the no response errors disappeared.

**Comment by werthdavid** [comment]
> FYI, I installed homebridge inside ubuntu wsl on Windows and getting the same issue...
> 
> Anyone? Anything???

I finally managed installing homebridge with WSL (Ubuntu on Windows 10).
config.json:

```json
{
 "mdns": { "interface": "192.168.2.103"}, // <-- HERE
"bridge": {
    "name": "HomebirdgeWSL",
    "username": "CC:AB:3A:E2:CD:32",
    "port": 51826,
    "pin": "313-44-555"
  },
  "description": "Homebridge Example",
```

where, of course, interface has the value of the Windows machines network interface.

**Comment by tomhoplon** [comment]
This issue rumbles on for me. Following a homebridge reset (often with a delete of the persist folder files), I re-add homebridge without a hitch but then all devices move to "not responding", sometimes within minutes, sometimes longer (occasionally as long as a week). And then a new reset is required.

I believe the cause may actually lie in the Apple system, specifically, when I don't use my Apply TV as a home hub (and Homepod does this job), it appears to be more stable. Has anyone else seen this?

**Comment by Stevy92** [comment]
I still use my Fritzbox and figured out, that avahi uses .local addresses. But avm uses fritz.box for local addresses and there is no way of changing it. So I installed dnsmasq on my raspberry pi and entered in the config file "raspberrypi.local = 192.168.178.2" and told my Fritzbox to use the raspberry as local dns server.
This solved the no responding on issue for now.
Hope it will work for longer.

**Comment by hepcat72** [comment]
For anyone else finding themselves here and trying these tips and it's not working... check your logs.  I was getting "no response" errors from Siri, couldn't control any device from the home app, couldn't re-add the homebridge to the home app, and none of the usual fixes worked either: remove persist, remove accessories, change the name, turn onn/off airplane mode, restart the Pi & my phone...  I even tried the dumptcp suggestion and saw homebridge traffic on the expected ports... Well, my homebridge-liftmaster2 plugin was crashing homebridge a dozen times a minute due to a bad response from the chamberlain remote server that the plugin doesn't handle.  My logs had bloated with multiple errors per second and pm2 was working overtime to restart homebridge continually.

**Comment by happy77** [comment]
My setup was running stable for years on same network. This now has started happening to me too, Alexa is still able to control all devices when Siri cannot. 

I have always used a local dns server on the rpi and except for upgrades of the rpi and homebridge / non nothing changed... any insight on what really causes this behavior change?

**Comment by a-x-** [comment]
I have "no response" too. I have to wait about minute  before  device becomes responsible
![UNADJUSTEDNONRAW_thumb_21ab](https://user-images.githubusercontent.com/6201068/64472095-c940a280-d161-11e9-8b98-ed801fcbb793.jpg)

**Comment by hepcat72** [comment]
Since I restored my phone after a trip to Mexico, I’ve been getting either no response or “couldn’t find anything like that” on a frequent and regular basis. Some iDevices completely refuse to control Homebridge devices and other iDevices insist that I append “outlet” or “switch” to the device name - but only for specific devices. It’s a royal mess and I believe it’s either Apple’s fault or the result of a mixed iOS environment. What I’d like to know is what tools can I use to determine where the breakdown in communication is?

**Comment by inikolaev** [comment]
> > FYI, I installed homebridge inside ubuntu wsl on Windows and getting the same issue...
> > Anyone? Anything???
> 
> I finally managed installing homebridge with WSL (Ubuntu on Windows 10).
> config.json:
> 
> ```json
> {
>  "mdns": { "interface": "192.168.2.103"}, // <-- HERE
> "bridge": {
>     "name": "HomebirdgeWSL",
>     "username": "CC:AB:3A:E2:CD:32",
>     "port": 51826,
>     "pin": "313-44-555"
>   },
>   "description": "Homebridge Example",
> ```
> 
> where, of course, interface has the value of the Windows machines network interface.

This resolved an issue for me while trying to run Homebridge on Raspberry Pi.

**Comment by mrdc** [comment]
> I have "no response" too. I have to wait about minute before device becomes responsible

The same issues for me after I've installed Homebridge Mi Humidifier plugin. Before it was fine.

**Comment by mrdc** [comment]
BTW Looks like it caused by an old unused bridges in persistent folder. Deleting them improves speed in my case.

**Comment by harveyappleton** [comment]
It may be of some help to someone - but I a few months ago swapped my home network infrastructure to be 2x Netgear switches (24 port and 8 port), and was having no end of issues for months. Couldn’t trace the problem but last night spotted that both switches had IGMP snooping turned ON, which after turning off, has made my homebridge installation so much more reliable and fast. 

Perhaps there should be a “recommended network settings / environment” page to help avoid issues like this? :)

**Comment by mafyata** [comment]
I use apple airport extreme 6th and igmp snooping has no effect. It is working good when it is on and off.

**Comment by hepcat72** [comment]
I'd like to provide some tips that can mitigate this issue in some cases so as to make it less frequent.  I discovered that sometimes, I was getting the error responses when my Pi showed 50% cpu usage (which is really 100% of one of the dual cores).  The culprit in this case was my pilight server.  It would get busy from time to time churching away on processing background RF signals.  I disabled the receiver (because I was only using it to record new remote buttons) and the incidence of the Siri error responses went way down.  It didn't completely go away, but not only did it become much less frequent, it also became much faster when it did work.  To see if this is an issue, do a top when Siri starts throwing errors and see if the pi is busy doing something else.

**Comment by snowdd1** [comment]
After a long and successful uptime, even surviving a move between two raspberries, it suddenly happened to me when switching to systemd-networkd for network card management (for getting a reliable LAN/WLAN failover). Now I have to remove/re-add 5 homebridge instances with hundreds of devices :-(

**Comment by mbicca** [comment]
Started having issues recently as well ...

Homebridge status says "not responding"

<img width="800" alt="image" src="https://user-images.githubusercontent.com/1126166/96553125-8ceadb80-1269-11eb-9514-3b49e62b7df6.png">

Tried pretty much ALL releases from 1.1.5 up to 1.3.0-beta-13

Previous configuration worked perfectly for a long time, it seems the issue started after updating nodejs to 12.19.0

Tried also a fresh install on Ubuntu 20.04, with new config file no luck, same message.

Everything "seems" to be up and running, don't really see any errors in the debug logs.
root@linux-atom1:/var/lib/homebridge# node -v
v12.19.0
root@linux-atom1:/var/lib/homebridge# systemctl status homebridge.service 
● homebridge.service - Homebridge
     Loaded: loaded (/etc/systemd/system/homebridge.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2020-10-20 00:14:20 PDT; 4min 30s ago
    Process: 32879 ExecStartPre=/bin/run-parts /etc/hb-service/homebridge/prestart.d (code=exited, status=0/SUCCESS)
    Process: 32882 ExecStartPre=/usr/lib/node_modules/homebridge-config-ui-x/dist/bin/hb-service.js before-start $HOMEBRIDGE_OPTS (code=exited, status=0/SUCCESS)
   Main PID: 32889 (node)
     CGroup: /system.slice/homebridge.service
             ├─32889 hb-service
             └─33161 homebridge

Oct 20 00:14:19 linux-atom1 systemd[1]: Starting Homebridge...
Oct 20 00:14:19 linux-atom1 run-parts[32880]: Ensuring /var/lib/homebridge is owned by homebridge
Oct 20 00:14:20 linux-atom1 systemd[1]: Started Homebridge.
Oct 20 00:14:20 linux-atom1 hb-service.js[32889]: [10/20/2020, 12:14:20 AM] [HB Supervisor] Homebridge Storage Path: /var/lib/homebridge
Oct 20 00:14:20 linux-atom1 hb-service.js[32889]: [10/20/2020, 12:14:20 AM] [HB Supervisor] Homebridge Config Path: /var/lib/homebridge/config.json
Oct 20 00:14:20 linux-atom1 hb-service.js[32889]: [10/20/2020, 12:14:20 AM] [HB Supervisor] Logging to /var/lib/homebridge/homebridge.log
Oct 20 00:17:21 linux-atom1 sudo[33139]: homebridge : TTY=pts/1 ; PWD=/usr/lib ; USER=root ; COMMAND=/usr/bin/npm install homebridge@1.3.0-beta.13
Oct 20 00:17:21 linux-atom1 sudo[33139]: pam_unix(sudo:session): session opened for user root by (uid=0)
Oct 20 00:17:32 linux-atom1 sudo[33139]: pam_unix(sudo:session): session closed for user root
root@linux-atom1:/var/lib/homebridge# hb-service status
ℹ Testing hb-service is running on port 8581...
✖ Homebridge UI Not Running
root@linux-atom1:/var/lib/homebridge# hb-service start
ℹ Starting Homebridge Service...
✔ Homebridge Started
**root@linux-atom1:/var/lib/homebridge# hb-service status
ℹ Testing hb-service is running on port 8581...
✖ Homebridge UI Not Running
root@linux-atom1:/var/lib/homebridge# ps -ef | grep -i home
homebri+   32889       1  5 00:14 ?        00:00:17 hb-service
homebri+   33161   32889 17 00:17 ?        00:00:16 homebridge**

I did notice that hb-service seems to start up fine but then when I run hb-service status it says UI not responding.

Is anyone having similar experiences right now?

Thanks,
Marco

**Comment by jnpatrick99** [comment]
I started to experience the same after Node upgrade to 14.15.1. Tried everything, nothing helps.
Tried removing Homebridge from accessories in Home App - now I lost all accessories and can't connect.

It's Windows, all ports are open, Bonjour is working.

**Comment by jimhosmer** [comment]
OK. So I've now deteriorated from a very nice Homekit/Homebridge/Homeseer installation that worked beautifully, through the irritating "no response" issue, to now being completely unable to reinstall Homebridge in Homekit. 

I'm on HomeBridge 1.1.7 on Windows 10. I removed Homebridge from Homekit due to the "no response" issue on all accessories and now I get an "accessory not found" response when I try to add Homebridge to Homekit.

Homebridge is still working fine with Alexa and I can control accessories via the Homebridge UI, so Homebridge/Homeseer are working very nicely.

Peak frustration.

**Comment by skywalker1979** [comment]
i've had a similar behaviour with a friend...
i managed to solve it by updating one of the npm modules...i think it was related to encryption.....
i'm not at home right now so i can't check....
i'll try to check what it was later on.

**Comment by Supereg** [comment]
@jimhosmer `"accessory not found"` typically means, that HomeKit unpaired the accessory (while it is offline) but the accessory doesn't know that it was unpaired, thus thinks it is still paired, and thus not showing in the list of available accessories.
Try to reset the pairing of your homebridge installation (either through Homebridge UI or by deleting the persist folder) followed by a restart.

**Comment by jimhosmer** [comment]
Thanks, I'll try these recommendations.

**Comment by jimhosmer** [comment]
OK, Reset worked. Accessories all added to Home app. Now let's see how long it all stays connected.
Thanks.

**Comment by a-x-** [comment]
HB is a bad thing. Try home assistant. It’s a better a bit and it have HomeKit integration too

**Comment by a-x-** [comment]
I will never back to poor laggish hb ever
